src/project_manager/ros_package_wizard.cpp

```diff
@@ -224,12 +224,42 @@ ROSPackageWizard::ROSPackageWizard()
     setFlags(Core::IWizardFactory::PlatformIndependent);
 }
 
+QDir ROSPackageWizard::getRootWorkspacePath(const QDir &path) const
+{
+    QFileInfoList list = QDir(path).entryInfoList(QStringList("*.workspace"));
+
+    if(list.count() == 0)
+    {
+        QDir parent(path);
+
+        if(parent.cdUp())
+            return getRootWorkspacePath(parent);
+    }
+
+    return path;
+}
+
 Core::BaseFileWizard *ROSPackageWizard::create(QWidget *parent,
                                                    const Core::WizardDialogParameters &parameters) const
 {
     Q_UNUSED(parameters);
     m_wizard = new ROSPackageWizardDialog(this, parent);
-    m_wizard->setPath(parameters.defaultPath());
+
+    QString defaultPath = parameters.defaultPath();
+
+#ifdef CREATE_FOLDERS
+    QDir workspaceRootPath = getRootWorkspacePath(QDir(defaultPath));
```

Comment:
I did looked at getWorkspaceInfo initially but do not see where I can get access to buildSystem and rosDistribution to use this function. This was my original idea. Can you point me to where and how to get them in ROSPackageWizard::create(). I will be glad to change it